### PR TITLE
Speed up FV tests

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Various FV speedups:

- Start felix instances and add routes in parallel.
- Do k8s cleanup in parallel.
- Set the QPS on our kube clients (and more importantly the one used by the calico client).

## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: fv/infrastructure/infra_k8s.go [hunk 1]
```diff
@@ -400,22 +411,37 @@ func (kds *K8sDatastoreInfra) Stop() {
...
-	cleanupAllHostEndpoints(kds.calicoClient)
-	cleanupAllFelixConfigurations(kds.calicoClient)
-	cleanupAllServices(kds.K8sClient)
+	startTime := time.Now()
+
+	var wg sync.WaitGroup
+	for _, f := range []cleanupFunc{
+		cleanupAllPods,
+		cleanupAllNodes,
+		cleanupAllNamespaces,
```
Comment (tomastigera) [inline fv/infrastructure/infra_k8s.go:424]: I wonder if the order completely does not matter. Can you clean up namespace before you clean up pods in it? Or does it get blocked until the other colliding thing completes?

File: fv/infrastructure/infra_k8s.go [hunk 2]
```diff
@@ -223,6 +230,9 @@ func setupK8sDatastoreInfra() (*K8sDatastoreInfra, error) {
 		if err == nil {
 			break
 		}
+		if strings.Contains(err.Error(), "already exists") {
```
Comment (neiljerram) [inline fv/infrastructure/infra_k8s.go:233]: How is this related?  (Maybe add a comment?)

File: fv/infrastructure/infra_k8s.go (hunk 1 above)
Comment (neiljerram) [reply fv/infrastructure/infra_k8s.go:424]: I guess there may be *some* blocking, but we still benefit by parallelising for cleanups that are independent.

File: fv/infrastructure/topology.go [hunk 3]
```diff
@@ -243,18 +259,23 @@ func StartNNodeTopology(n int, opts TopologyOptions, infra DatastoreInfra) (feli
...
+				jBlock := fmt.Sprintf("10.65.%d.0/24", j)
+				if opts.IPIPEnabled && opts.IPIPRoutesEnabled {
+					err := iFelix.ExecMayFail("ip", "route", "add", jBlock, "via", jFelix.IP, "dev", "tunl0", "onlink")
+					Expect(err).ToNot(HaveOccurred())
+				} else if opts.VXLANMode == api.VXLANModeNever {
+					// If VXLAN is enabled, Felix will program these routes itself.
+					err := iFelix.ExecMayFail("ip", "route", "add", jBlock, "via", jFelix.IP, "dev", "eth0")
+					Expect(err).ToNot(HaveOccurred())
+				}
+			}(i, j, iFelix, jFelix)
```
Comment (neiljerram) [inline fv/infrastructure/topology.go:275]: I'm surprised if there is a lot of time saving here.  Have you measured this one?

Comment (fasaxc) [reply fv/infrastructure/topology.go:275]: Yes, it was a few hundred millis IIRC; seemed worth it given we have 1000+ tests.

File: fv/infrastructure/infra_k8s.go (hunk 1 above)
Comment (fasaxc) [reply fv/infrastructure/infra_k8s.go:424]: Yes, deleting a namespace deletes all pods inside by the garbage collector.  But that's slower.
